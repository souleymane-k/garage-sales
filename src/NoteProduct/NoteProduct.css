.Result {
    background-color: #EAF6F6;
    /* background-image: url(../../souleymane-3.jpg); */
    color: black;
    margin: 0px auto;
    max-width: 280px;
    padding: 8px 0 8px 8px;
    border: solid;
    border-radius: 10px;
    /* position: relative; */
  }
  h2{
      text-align: center;
      /* color:white; */
  }
  
  .Result_link{
    background-color: #bd0909;
    margin:5px;
    text-align:center;
  }
  .Result_link:hover {
    color: white;
  }
  .Result_delete:hover {
    background-color: #bd0909;
    color: white;
    cursor: pointer;
    font-weight: bold;
  }
  .resultElements{
    /* position:absolute; */
    /* margin: 1px auto; */
    padding:5px;
    height:auto;
    margin:2px;
    /* margin:50px auto; */
  }
  .Result_month_taken{
    background-color: greenyellow;
    margin:5px;
    padding:2px;
    border-radius: 10px;
    text-align:center;
  }
  .Result__meal_taken{
    background-color: white;
    margin:5px;
    padding:2px;
    border-radius: 10px;
    text-align:center;
  }
  .Result_result_read{
    background-color: #eba83a;
    margin:5px;
    /* padding:2px; */
    border-radius: 10px;
    text-align:center;
  }
  .Result_month_id{
    background-color: greenyellow;
    margin:5px;
    padding:2px;
    border-radius: 10px;
    text-align:center;
  }
  .Results_description{
    background-color: #ffc7c7;
    margin:5px;
    /* padding:2px; */
    border-radius: 10px;
    text-align:center;
  }
  .Result_userid{
    background-color: #eba83a;
    margin:5px;
    /* padding:2px; */
    border-radius: 10px;
    text-align:center;
  }
  .Result_diabetestype{
    background-color: #00ead3;
    margin:5px;
    /* padding:2px; */
    border-radius: 10px;
    text-align:center;
  }
  .Result_delete{
    background-color: red;
    margin:10px auto;
    width:100%;
    height: 30px;
    /* padding:2px; */
    /* border-radius: 10px; */
    text-align:center;
    border:none;
  }
  .edit_product{
    background-color:greenyellow;
    margin:10px auto;
    width:100%;
    height: 30px;
    text-align:center;
    border:none;
  }
  /* .result-form{
    background-image: url(../../souleymane-3.jpg);
  } */
  span{
    margin-left: 10px;
  }
  

  
  @media all and (min-width: 640px) {

    .Result {
      /* background-color: #EAF6F6; */
      /* background-image: url(../../souleymane-3.jpg); */
      color: black;
      margin: 15px auto;
      max-width: 40%;
      padding: 8px 0 8px 8px;
      border: solid;
      border-radius: 10px;
      /* position: relative; */
    }
    .Result_result_dates{
      background-color: #00ead3;
      margin:5px;
      /* padding:2px; */
      border-radius: 10px;
      text-align:center;
    }
    /* .result-form{
      background-image: url(../../souleymane-3.jpg);
    } */
  }