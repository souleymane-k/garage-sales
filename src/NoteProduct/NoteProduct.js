import React from 'react'
import {  Link } from 'react-router-dom'
import ApiContext from '../contexts/ApiContext';
import TokenService from '../services/TokenService'
import config  from '../config.js'
import './NoteProduct.css'
import moment from 'moment'

export default class NoteProduct extends React.Component {
  static defaultProps ={
    onDeleteProduct: () => {},
  }
  static contextType = ApiContext;

  handleClickDelete = e => {
    e.preventDefault()
    const id = this.props.id

    fetch(`${config.API_ENDPOINT}/products/${id}` , {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json',
        'authorization':`bearer ${TokenService.getAuthToken()}`
      },
    })
      .then(() => {
        this.props.onDeleteProduct(id)
      })
      .catch(error => {
        console.log(e)
        console.error({ error })
      })
  }


  render() {
    
    const {id,product_name, product_price, date_posted,description} = this.props
    
    return (
      <div className="result-form">
      <div className='Result'>
        <div className="resultElements">
        <div className='Result_month_taken'>
          <div className='Result_results_description'>
              name:
          {'  '}
            <span className='month_taken'>
              {product_name}
            </span> 
          </div> 
        
         </div>
         <div className='Result__meal_taken'>
          <div className='Resultresults_meal_taken'>
             price:
          {'   '}
            <span className='meal_taken'>
              {product_price}
            </span> 
          </div> 
         </div>
         <div className='Result_result_dates'>
          <div className='Result_result_date__tested'>
              date:
          {'   '}
            <span className='date_tested'>
            {moment.utc(date_posted).format("MM/DD/YYYY")}
              
            </span> 
          </div> 
         </div>

        
         <div className='Results_description'>
          <div className='Result_results_description'>
           Product Description:
            {'   '}
            <span className='Description'>
              {description}
            </span> 
          </div> 
         </div>
         <button
          className='Result_delete'
          type='button'
          onClick={this.handleClickDelete}
        >
          DELETE
        </button>
        <Link  to={`/editproduct/${id}`} className="edit_product">Edit Product</Link>
      </div>
      </div>
      </div>

    )
  }
}