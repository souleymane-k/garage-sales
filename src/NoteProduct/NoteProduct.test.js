import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteProduct from './NoteProduct';
import {MemoryRouter} from 'react-router-dom'

describe(`NoteProduct component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<MemoryRouter><NoteProduct /></MemoryRouter>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter><NoteProduct /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})