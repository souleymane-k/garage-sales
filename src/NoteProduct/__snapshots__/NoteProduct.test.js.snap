// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteProduct component renders the complete form 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "ht0jv7",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "ht0jv7",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NoteProduct
    onDeleteProduct={[Function]}
  />
</Router>
`;
